packages/core/src/builtin-tasks/contract.ts

```diff
@@ -198,6 +210,181 @@ subtask('contract:deploy')
+        var: name,
+      })
+    }
+
+    const conrtactBinPath = await (async () => {
+      // check bin path is absolute path
+      if (binPath && path.isAbsolute(binPath)) {
+        return binPath
+      }
+
+      if (binPath) {
+        return path.join(process.cwd(), binPath)
+      }
+
+      const workspace = (await run('contract:get-workspace')) as string
+
+      return path.join(workspace, `build/release/${name}`)
+    })()
+
+    if (!existsSync(conrtactBinPath)) {
+      throw new KuaiError(ERRORS.BUILTIN_TASKS.CONTRACT_RELEASE_FILE_NOT_FOUND, {
+        var: name,
+      })
+    }
+
+    const lumosConfig: Config = {
+      PREFIX: ckbChain.prefix,
+      SCRIPTS: ckbChain.scripts || {
+        ...(await getGenesisScriptsConfig(ckbChain.rpcUrl)),
+      },
+    }
+
+    const messageSigner: MessageSigner = (message, fromInfo) => {
+      const { multisigScript } = parseFromInfo(fromInfo, { config: lumosConfig })
+      return run('contract:sign-message', {
+        message,
+        address: encodeToAddress(parseFromInfo(fromInfo, { config: lumosConfig }).fromScript, { config: lumosConfig }),
+        signer,
+        prefix: multisigScript,
+      }) as Promise<string>
+    }
+
+    const contractDeployer = new ContractDeployer(
+      {
+        rpcUrl: config.ckbChain.rpcUrl,
+        config: lumosConfig,
+      },
+      signer ? messageSigner : undefined,
+    )
+
+    const deployTx = await contractDeployer.upgrade(
+      conrtactBinPath,
+      deployerInfos[0],
+      feepayerInfos[0],
+      {
+        txHash: migration.tx_hash,
+        index: migration.index,
+        dataHash: migration.data_hash,
+      },
+      { feeRate },
+    )
+    const { tx } = deployTx
+
+    if (args.export) {
+      const exportPath = path.isAbsolute(args.export) ? args.export : path.join(process.cwd(), args.export)
+
+      const exportData = {
+        transaction: ParamsFormatter.toRawTransaction(tx),
+        multisig_configs: {},
+        signatures: {},
+      }
+
+      const infos = [...feepayerInfos, ...deployerInfos]
+      infos.forEach((fromInfo) => {
+        if (isMultisigFromInfo(fromInfo)) {
+          const { fromScript } = parseFromInfo(fromInfo, { config: lumosConfig })
+          const template = lumosConfig.SCRIPTS['SECP256K1_BLAKE160']!
+          Object.assign(exportData.multisig_configs, {
+            [fromScript.args]: {
+              sighash_addresses: fromInfo.publicKeyHashes.map((args) =>
+                scriptToAddress(
+                  {
+                    codeHash: template.CODE_HASH,
+                    hashType: template.HASH_TYPE,
+                    args: args,
+                  },
+                  { config: lumosConfig },
+                ),
+              ),
+              require_first_n: fromInfo.R,
+              threshold: fromInfo.M,
+            },
+          })
+        }
+      })
+
+      writeFileSync(exportPath, JSON.stringify(exportData, null, 2))
```

Comment:
In what case will it be exported and sent simultaneously?

Reply:
Is this just an option for the user, or how about sending without exporting by default?

Reply:
IMO, either `send` or `export` is enough because I cannot imagine a case of `send & export`

Reply:
Any update on this conversation

Reply:
ok I'll update it to send by default in the absence of export